src/components/Widget/IntegrationsWidget.tsx

```diff
@@ -0,0 +1,477 @@
+        red_hat: 0,
+        openshift: 0,
+      };
+  
+      const redHatSourceTypes: Record<string, keyof typeof counts> = {
+        '1': 'openshift',
+      };
+  
+      counts.red_hat = data.meta.count;
+  
+      data.sources.forEach((source: { source_type_id: string }) => {
+        const sourceKey = redHatSourceTypes[source.source_type_id];
+        if (sourceKey) {
+          counts[sourceKey]++;
+        }
+      });
+    
+      setIsLoading(false);
+      setIntegrationCounts((prevCounts) => ({ ...prevCounts, ...counts }));
+    } catch (error) {
+      console.error('Unable to get Red Hat Sources', error);
+    }
+  };
+    
+
+  const fetchIntegrations = async () => {
+    try {
+      const response = await fetch('/api/integrations/v1.0/endpoints?type=ansible&type=webhook&type=camel');
+      const { data } = await response.json();
+
+      const validCamelSubTypes = ['teams', 'google_chat', 'slack', 'servicenow', 'splunk'];
+
+      const counts: { [key: string]: number } = {
+        ansible: 0,
+        webhook: 0,
+        camel: 0,
+        camel_google_chat: 0,
+        camel_teams: 0,
+        camel_slack: 0,
+        camel_servicenow: 0,
+        camel_splunk:0
+      };
+
+      counts.ansible = data.filter((integration: Integration) => integration.type === 'ansible').length;
+      counts.webhook = data.filter((integration: Integration) => integration.type === 'webhook').length;
+      data.forEach((integration: Integration) => {
+        if (integration.type === 'camel') {
+          counts.camel++;
+          if (integration.sub_type) {
+            counts[integration.sub_type] = (counts[integration.sub_type] || 0) + 1;
+          }
+        }
+      });
+
+      const camelSubTypeCount = data.filter(
+        (integration: Integration) => integration.type === 'camel' && validCamelSubTypes.includes(integration.sub_type ?? '')
+      ).length;
+      counts.camel += camelSubTypeCount;
+
+      setIsLoading(false);
+      setIntegrationCounts((prevCounts) => ({ ...prevCounts, ...counts }));
+    } catch (error) {
+      console.error('Unable to get Integrations ', error);
+    }
+  };
+
+  useEffect(() => {
+    setIsLoading(true);
+    fetchCloudSources();
+    fetchRedHatSources();
+    fetchIntegrations();
+  }, []);
+
+  const badgeCounts = (items: typeof integrationsData[0]['items']) => {
+    return items.reduce((total, item) => total + (integrationCounts[item.id] || 0), 0);
+  };
+
+  const isEmptyState = Object.values(integrationCounts).reduce((total, count) => total + count, 0) === 0;
+
+  return (
+    <>
+      {isLoading ? (
+        <Spinner />
+      ) : isEmptyState ? (
+        <>
+          <Card isPlain ouiaId="integrations-widget-empty-state" isClickable>
+            <CardBody>
+              Click on a third-party application to create an integration for it.{' '}
+              <a href="/settings/integrations?category=overview">Learn more about Integrations.</a>
```

Comment:
This would refresh the application, we should probably use `Link` component so we don't refresh the entire UI.